vpr/src/route/rr_graph_storage.cpp

```diff
@@ -549,19 +549,19 @@ const char* t_rr_graph_view::node_type_string(RRNodeId id) const {
     return rr_node_typename[node_type(id)];
 }
 
-const char* t_rr_graph_storage::node_direction_string(RRNodeId id) const {
-    e_direction direction = node_direction(id);
+const std::string& t_rr_graph_storage::node_direction_string(RRNodeId id) const {
+    Direction direction = node_direction(id);
 
-    if (direction == e_direction::INC_DIRECTION) {
-        return "INC_DIR";
-    } else if (direction == e_direction::DEC_DIRECTION) {
-        return "DEC_DIR";
-    } else if (direction == e_direction::BI_DIRECTION) {
-        return "BI_DIR";
+    if (direction == Direction::INC) {
```

Comment:
I'd rewrite this to get rid of the if's.  The usual way to do this is something like:

int_direction = static_cast<int>(direction);
VTR_ASSERT (int_direction >= 0 && int_direction < Direction::NUM_DIRECTIONS);
return CONST_DIRECTION_STRING[int_direction];

Shorter and scales better if we add more enum values later.

Reply:
Yeah, that makes sense. I'll adjust that real quick.

Reply:
Just noticed by syntax is bad:  meant 
int_direction = static_cast<int>(direction); 

and so on :).

Reply:
switch/case ?